src/iceberg_explorer/api/routes/catalog.py

```
@@ -0,0 +1,458 @@
+    if not namespace_parts:
+        raise HTTPException(
+            status_code=400,
+            detail="Namespace cannot be empty",
+        )
+
+    catalog_name = engine.catalog_name
+
+    with engine.get_connection() as conn:
+        namespace_path = _build_namespace_path(catalog_name, namespace_parts)
+        try:
+            conn.execute(f"SELECT * FROM {namespace_path}.information_schema.schemata LIMIT 1")
```

Ruff (0.14.10)

Possible SQL injection vector through string-based query construction

(S608)